crypto/fipsmodule/cmac/cmac.c

```diff
@@ -83,6 +83,12 @@ static void CMAC_CTX_cleanup(CMAC_CTX *ctx) {
 
 int AES_CMAC(uint8_t out[16], const uint8_t *key, size_t key_len,
              const uint8_t *in, size_t in_len) {
+  // We have to verify that all the CMAC services actually succeed before
+  // updating the indicator state, so we lock the state here.
+  FIPS_service_indicator_lock_state();
+  CMAC_CTX *ctx = OPENSSL_malloc(sizeof(*ctx));
```

Comment:
Why was it needed to have `ctx` in the heap and not on the stack? Does it make the service indicator update fail?

Reply:
There were some old compiler failures complaining about `ctx`  may be uninitialized in this function. It's most likely because of the new `goto` statements inserted within the function, but nothing functionality-wise was affected.

Reply:
But now the change is causing some unknown failures in address sanitizers, I'm probably going to be looking to reverse the entire `goto` change.

Reply:
ACK